src/terraform-resource/terraform/planfile.go

```diff
@@ -0,0 +1,78 @@
+package terraform
```

Comment:
I'm not saying we should get rid of `StateFile` and `PlanFile`, but we could extract the common logic into a shared struct that is not exported from the package. We could then use something like [Struct composition](http://dave.cheney.net/2015/05/22/struct-composition-with-go) to embed the shared struct.